app/src/main/java/com/peluso/walletguru/viewmodel/MainViewModel.kt

```diff
@@ -39,8 +40,12 @@ class MainViewModel : ViewModel() {
         //TODO: make this based on actual values and standings of the accounts. We should probably
         // make this a dynamically calculated ranking based on your input (which should use the same viewmodel)
         // for now I'll make it loop through all of them and add it to the viewstate
-        val reduced = map.values.reduce { acc, list -> acc + list }
-        _viewState.postValue(_viewState.value?.copy(submissions = reduced, isLoading = false))
+        //val reduced = map.values.reduce { acc, list -> acc + list }
+        //_viewState.postValue(_viewState.value?.copy(submissions = reduced, isLoading = false))
+
+        _viewState.value?.userAccounts.let {
+            _viewState.postValue(_viewState.value?.copy(submissions = it?.orderSubmissions(map)))
```

Comment:
also, be sure to add the `isLoading = false` to the viewstate change here. This will tell our UI to stop showing the loading bar.

Reply:
Added it on my most recent push.